### DISCUSSION TITLE
Better Keyboards for KOReader!

### DISCUSSION THREAD

**Discussion by sethhammett** [koreader/koreader#7951] [Show and tell]
Recently I've been making some huge improvements to the keyboards in KOReader, like adding numbers on the top and arrow keys. "Umlaut mode" has been replaced with "Alt mode" since you can access all the umlauts you need in the popup menu for a key. "Alt mode" replaces letters and symbols with alternate versions, for example > with ≥ and - with —. I fixed a lot of clutter in the popup menus too, such as the random blank spaces and symbols in the popup for the K key. "Sym" has been renamed to ⌘. And wait, is that a tab key?!?!

There are three keyboards included in the attached file: Touch, Regular, and Full.

 * The Touch keyboard has as few keys as possible so that the keys can be bigger. Use this one if you want to type something really quickly, like taking notes. It is the first shown in the attached image.
 * The Regular keyboard has a balance between symbols and functionality, and personally is my favorite. Use this one if the Touch keyboard is too hard to navigate and the Full keyboard is too tiny. It is the second shown in the attached image.
 * The Full keyboard has lots of symbols shown, it more closely resembles the keyboard you are using right now. This one is the fastest to type on but expect a lot of typos. Use this one for coding and ASCII art. It is the third shown in the attached image.

Here's the file: [kbk.zip (27 KB)](https://github.com/koreader/koreader/files/6793973/kbk.zip). If you have a Kobo just extract the contents into your reader's main directory, I don't know about Kindle or anything else, but the process should be similar. Make sure to enable the layouts in Settings > Device > Keyboard layout. You can toggle between the layouts with the 🌐 button. Let me know if anything doesn't work. All the original files are in there renamed to *_old.

Some info I forgot to add earlier:
 * Swiping keys to type accents/symbols has been disabled because when I try to type fast it just types a bunch of accents and stuff.
 * In "Night Mode" the keys appear dark grey with a black outline.

![Keyboards](https://user-images.githubusercontent.com/83931313/125134916-61dd3c00-e0c5-11eb-81f0-7faf16878aeb.jpg)

**Comment by hius07** [comment]
Why haven't you located capital letters right above the lower ones in the popup windows?
It would allow entering the capitals with just swipe up for all letters.

**Comment by poire-z** [comment]
Had a quick try, and looks ok enough to be discussed seriously :) (an it already has 2 :+1:)

~~Can you please open an issue or a PR, copy your first post as-is, so it's more visible and we can discuss it better.~~
~~Gonna try the `Create issue from discussion` button.~~
No, I'd rather let you the original author do it. Do you have  `Create issue from discussion` in the right bar ?

Because GitHub "discussions" are just not good for serious discussions.

**Comment by poire-z** [comment]
Please continue discussion only in ~~https://github.com/koreader/koreader/pull/7960~~ https://github.com/koreader/koreader/pull/7972
